src/rgw/rgw_http_errors.h

```diff
@@ -10,65 +10,66 @@ struct rgw_http_errors {
+    { ERR_BAD_DIGEST, 400, "BadDigest", "The Content-MD5 you specified did not match what we received." },
+    { ERR_INVALID_BUCKET_NAME, 400, "InvalidBucketName", "The specified bucket is not valid." },
+    { ERR_INVALID_OBJECT_NAME, 400, "InvalidObjectName", "The specified object is not valid." },
+    { ERR_UNRESOLVABLE_EMAIL, 400, "UnresolvableGrantByEmailAddress", "The email address you provided does not match any account on record." },
+    { ERR_INVALID_PART, 400, "InvalidPart", "One or more of the specified parts could not be found. The part might not have been uploaded, or the specified entity tag might not have matched the part's entity tag." },
+    { ERR_INVALID_PART_ORDER, 400, "InvalidPartOrder", "The list of parts was not in ascending order.Parts list must specified in order by part number." },
+    { ERR_REQUEST_TIMEOUT, 400, "RequestTimeout", "Your socket connection to the server was not read from or written to within the timeout period." },
+    { ERR_TOO_LARGE, 400, "EntityTooLarge", "Your proposed upload exceeds the maximum allowed object size." },
+    { ERR_TOO_SMALL, 400, "EntityTooSmall", "Your proposed upload is smaller than the minimum allowed object size." },
+    { ERR_TOO_MANY_BUCKETS, 400, "TooManyBuckets", "You have attempted to create more buckets than allowed." },
+    { ERR_MALFORMED_XML, 400, "MalformedXML", "This happens when the user sends malformed xml (xml that doesn't conform to the published xsd) for the configuration. The error message is, \"The XML you provided was not well-formed or did not validate against our published schema.\"" },
+    { ERR_AMZ_CONTENT_SHA256_MISMATCH, 400, "XAmzContentSHA256Mismatch", "Amazon contents sha256 mismatch"},
+    { ERR_LENGTH_REQUIRED, 411, "MissingContentLength", "You must provide the Content-Length HTTP header." },
+    { EACCES, 403, "AccessDenied", "Access Denied" },
+    { EPERM, 403, "AccessDenied", "Access Denied" },
+    { ERR_SIGNATURE_NO_MATCH, 403, "SignatureDoesNotMatch", "The request signature we calculated does not match the signature you provided. Check your AWS secret access key and signing method. For more information, see REST Authentication and SOAP Authentication for details." },
+    { ERR_INVALID_ACCESS_KEY, 403, "InvalidAccessKeyId", "The AWS access key Id you provided does not exist in our records." },
+    { ERR_USER_SUSPENDED, 403, "UserSuspended", "User Suspended" },
+    { ERR_REQUEST_TIME_SKEWED, 403, "RequestTimeTooSkewed", "The difference between the request time and the server's time is too large." },
+    { ERR_QUOTA_EXCEEDED, 403, "QuotaExceeded", "Quota Exceeded" }, 
+    { ENOENT, 404, "NoSuchKey", "The resource you requested does not exist" },
+    { ERR_NO_SUCH_BUCKET, 404, "NoSuchBucket", "The specified bucket does not exist." },
+    { ERR_NO_SUCH_WEBSITE_CONFIGURATION, 404, "NoSuchWebsiteConfiguration", "No such website configuration" },
```

Comment:
Hmm, it was correctly in place in Jewel, but seems to have been lost on Master:
https://github.com/ceph/ceph/blob/jewel/src/rgw/rgw_rest_s3.cc#L4251